docs/kb/wasm.rst

```diff
@@ -0,0 +1,307 @@
+=======================================
+Wasm support for user-defined functions
+=======================================
+
+This document describes the details of Wasm language support in user-defined functions (UDF). The language ``wasm`` is one of the possible languages to use, besides Lua, to implement these functions.
+
+
+.. note:: Until Scylla 5.2, the Wasm language is called ``xwasm``. This name will be replaced with ``wasm`` in Scylla 5.3.
+
+How to generate a correct Wasm UDF source code
+----------------------------------------------
+
+Scylla accepts UDF's source code in WebAssembly text format - also known as ``WAT``. The source can use and define whatever's needed for execution, including multiple helper functions and symbols. The requirements for it to be accepted as correct UDF source is that the WebAssembly module exports a symbol with the same name as the function, this symbol's type is indeed a function with correct signature, the module exports a ``_scylla_abi`` global and all symbols related to the selected ABI version.
+
+UDF's source code can be, naturally, simply coded by hand in ``WAT``. It is not often very convenient to program directly in assembly, so here are a few tips.
```

Comment:
It would make sense to put an WAT example after mentioning that it can be written by hand, but we expect most users won't want to do anything with the WAT code, and just generate it using the available tools (the Rust helper library) instead. Because of that, I'd like to have that example as the first one on the page.